app/src/main/java/com/capstone/event_finder/fragments/ProfileFragment.java

```diff
@@ -24,6 +55,102 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        eventViewModel = new ViewModelProvider(this).get(EventViewModel.class);
+        tvProfileUsername = view.findViewById(R.id.tvProfileUsername);
+        rvBookmarks = view.findViewById(R.id.rvBookmarks);
+        tvProfileUsername.setText(ParseUser.getCurrentUser().getUsername());
+
+        JsonArray allBookmarks = new JsonArray();
+        setUpRecyclerView(view);
+        getAndSetUserBookmarks(allBookmarks);
+    }
+
+    public void getAndSetUserBookmarks(JsonArray allBookmarks) {
+        bookmarkList.clear();
+        userBookmarks.clear();
+        bookmarkIds.clear();
+        queryUserBookmarksFromParse(allBookmarks);
+    }
+
+    private void tryRetrieveEventInCache(String eventId, JsonArray allBookmarks) {
```

Comment:
Each method here on its own is good, and having this in the ProfileFragment is fine to proof of concept for the MVP.  However, in the end abstracting this into the ViewModel can help clean this up a bunch.  See my note in `EventViewModel.java`